src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -429,10 +429,19 @@ CitusModifyBeginScan(CustomScanState *node, EState *estate, int eflags)
 		PlanState *planState = &(scanState->customScanState.ss.ps);
 
 		ExecuteMasterEvaluableFunctions(jobQuery, planState);
+		ExecuteTaskListFunctions(taskList, planState);
```

Comment:
Shouldn't it be sufficient to evaluate functions in the `jobQuery` and then break down the jobQuery into multiple tasks?